src/red.ts

```diff
@@ -213,13 +214,16 @@ export const serializedRedEnvSourceText = (function redEnvFactory(blueEnv: Membr
     }
 
     function copyRedOwnDescriptors(shadowTarget: RedShadowTarget, blueDescriptors: PropertyDescriptorMap) {
-        for (const key in blueDescriptors) {
+        // using ownKeys here to get all string/symbol/number property names
+        const keys = ownKeys(blueDescriptors);
+
+        ArrayForEach(keys, (key: string | symbol | number) => {
```

Comment:
Would just a for-loop work here?

Reply:
YES! I'll change it to a for-loop, not as pretty syntax but probably better perf.

Reply:
Ya the ES5 functional composition functions can drag down perf of low level things esp when they get stacked.